lib/bloc/account/account_bloc.dart

```diff
@@ -331,4 +332,45 @@ class AccountBloc extends Cubit<AccountState> with HydratedMixin {
     _log.info("mnemonicsValidated");
     emit(state.copyWith(verificationStatus: VerificationStatus.VERIFIED));
   }
+
+  List<PaymentMinutiae> filterPaymentList() {
+    final nonFilteredPayments = state.payments;
+    final paymentFilters = state.paymentFilters;
+
+    var filteredPayments = nonFilteredPayments;
+    // Apply date filters, if there's any
+    if (paymentFilters.fromTimestamp != null || paymentFilters.toTimestamp != null) {
+      filteredPayments = nonFilteredPayments.where((paymentMinutiae) {
+        final fromTimestamp = paymentFilters.fromTimestamp;
+        final toTimestamp = paymentFilters.toTimestamp;
+        final milliseconds = paymentMinutiae.paymentTime.millisecondsSinceEpoch;
+        if (fromTimestamp != null && toTimestamp != null) {
+          return fromTimestamp < milliseconds && milliseconds < toTimestamp;
+        }
+        return true;
+      }).toList();
+    }
+
+    // Apply payment type filters, if there's any
+    if ((paymentFilters.filters != null && paymentFilters.filters != sdk.PaymentTypeFilter.values)) {
+      filteredPayments = filteredPayments.where((paymentMinutiae) {
+        for (var f in paymentFilters.filters!) {
+          if (f == sdk.PaymentTypeFilter.Sent && paymentMinutiae.paymentType == sdk.PaymentType.Sent) {
+            return true;
+          }
+          if (f == sdk.PaymentTypeFilter.ClosedChannels &&
+              paymentMinutiae.paymentType == sdk.PaymentType.ClosedChannel) {
+            return true;
+          }
+          if (f == sdk.PaymentTypeFilter.Received &&
+              paymentMinutiae.paymentType == sdk.PaymentType.Received) {
+            return true;
+          }
+          return false;
+        }
```

Comment:
This logic can be improved by using
`return paymentFilters.filters!.contains(paymentMinutiae.paymentType);`

Reply:
I now see that the `PaymentType` & `PaymentTypeFilter` aren't the same type, we can use

`return paymentTypeFilters.any((filter) => filter.name == paymentMinutiae.paymentType.name);`

However, there's a difference between `PaymentTypeFilter.ClosedChannels` & `PaymentType.ClosedChannel` which needs to be addressed on SDK.

Reply:
This isn't possible since the one is of type `PaymentTypeFilter`and the other is of type `PaymentType`. I think If we want this we have to override the comparable method.

Reply:
> This isn't possible since the one is of type `PaymentTypeFilter`and the other is of type `PaymentType`. I think If we want this we have to override the comparable method.

See the comment above. We need to rename PaymentTypeFilter.ClosedChannels to PaymentTypeFilter.ClosedChannel on the SDK and do the comparison via name of the enum.

Reply:
Understand, I will create a pr for this.